### ISSUE TITLE
App crashes when approving storage permission in Marshmallow

### ISSUE THREAD

**Issue by brad** [PaulWoitaschek/MaterialAudiobookPlayer#389]
I think I managed to send a crash report just now. The good news is that the app comes right back and is usable again after the crash. Tested in version 2.5.0.6

**Comment by PaulWoitaschek** [comment]
And where did you send it to?

**Comment by brad** [comment]
Oh, I just enabled "Report bugs" in preferences before reproducing the issue

**Comment by PaulWoitaschek** [comment]
And where did you get the version from?

**Comment by PaulWoitaschek** [comment]
If you are using a debug version it will not send crashes automatically.

The version I just hosted on dropbox logs to a file on your sd card named `materialaudiobookplayer.log` at the root.

**Comment by brad** [comment]
Sorry about that @PaulWoitaschek I think I actually sent it now. Subject is "for issue 389"

**Comment by PaulWoitaschek** [comment]
Didn't arrive here. Try <a href="mailto:woitaschek@posteo.de">this one</a>.

**Comment by brad** [comment]
Sorry Paul, the form doesn't allow inputting a destination email address. I assumed it could be found under the reports section in the Google Play developer console. I've emailed you a portion of adb logcat output containing the crash. I hope that gives enough to go on. Also, I have been reproducing the error in the app from the play store now with automatic bug reporting enabled, so maybe that is working now?

**Comment by PaulWoitaschek** [comment]
Thanks.

```logcat
01-05 08:57:03.022 16893 16893 E ACRA    : ACRA caught a RuntimeException for de.ph1b.audiobook
01-05 08:57:03.022 16893 16893 E ACRA    : java.lang.RuntimeException: Failure delivering result ResultInfo{who=@android:requestPermissions:, request=1, result=-1, data=Intent { act=android.content.pm.action.REQUEST_PERMISSIONS (has extras) }} to activity {de.ph1b.audiobook/de.ph1b.audiobook.view.FolderChooserActivity}: kotlin.KotlinNullPointerException
01-05 08:57:03.022 16893 16893 E ACRA    : 	at android.app.ActivityThread.deliverResults(ActivityThread.java:3699)
01-05 08:57:03.022 16893 16893 E ACRA    : 	at android.app.ActivityThread.handleSendResult(ActivityThread.java:3742)
01-05 08:57:03.022 16893 16893 E ACRA    : 	at android.app.ActivityThread.-wrap16(ActivityThread.java)
01-05 08:57:03.022 16893 16893 E ACRA    : 	at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1393)
01-05 08:57:03.022 16893 16893 E ACRA    : 	at android.os.Handler.dispatchMessage(Handler.java:102)
01-05 08:57:03.022 16893 16893 E ACRA    : 	at android.os.Looper.loop(Looper.java:148)
01-05 08:57:03.022 16893 16893 E ACRA    : 	at android.app.ActivityThread.main(ActivityThread.java:5417)
01-05 08:57:03.022 16893 16893 E ACRA    : 	at java.lang.reflect.Method.invoke(Native Method)
01-05 08:57:03.022 16893 16893 E ACRA    : 	at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:726)
01-05 08:57:03.022 16893 16893 E ACRA    : 	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:616)
01-05 08:57:03.022 16893 16893 E ACRA    : Caused by: kotlin.KotlinNullPointerException
01-05 08:57:03.022 16893 16893 E ACRA    : 	at de.ph1b.audiobook.presenter.FolderChooserPresenter.refreshRootDirs(FolderChooserPresenter.kt:216)
01-05 08:57:03.022 16893 16893 E ACRA    : 	at de.ph1b.audiobook.presenter.FolderChooserPresenter.gotPermission(FolderChooserPresenter.kt:72)
01-05 08:57:03.022 16893 16893 E ACRA    : 	at de.ph1b.audiobook.view.FolderChooserActivity.onRequestPermissionsResult(FolderChooserActivity.kt:109)
01-05 08:57:03.022 16893 16893 E ACRA    : 	at android.app.Activity.dispatchRequestPermissionsResult(Activity.java:6582)
01-05 08:57:03.022 16893 16893 E ACRA    : 	at android.app.Activity.dispatchActivityResult(Activity.java:6460)
01-05 08:57:03.022 16893 16893 E ACRA    : 	at android.app.ActivityThread.deliverResults(ActivityThread.java:3695)
01-05 08:57:03.022 16893 16893 E ACRA    : 	... 9 more
```

**Comment by PaulWoitaschek** [comment]
Fixed as of f4ad00c52e99cb56f67092f5f714e384c2780dbe. Thanks!

**Comment by brad** [comment]
Confirmed fixed!
